atmo/coordinator/capabilities/capabilities.go

```diff
@@ -67,7 +67,8 @@ func ResolveFromSource(source appsource.AppSource, ident, namespace, version str
 		defaultConfig.File = userConfig.File
 	}
 
-	if userConfig.DB != nil {
+	// Override the connections.Database struct
+	if userConfig.DB.Enabled {
```

Comment:
I believe the correct check would be `if userConfig.DB != nil && userConfig.DB.Enabled` so that we avoid nil pointer panics when `DB` is nil.